### PR TITLE
Add Jenkinsfile.deploy for daily builds

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by mareknovotny** [comment]
I guess we should change the version of project too or not? as we release 0.x and it is 8.0.0-SNAPSHOT we should synchronize it. WDYT?

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        buildDiscarder logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '', numToKeepStr: '5')
+        timeout(time: 90, unit: 'MINUTES')
+    }
+    stages {
+        stage('Initialize') {
+            steps {
+                sh 'printenv'
+            }
+        }
+        stage('Build kogito-bom') {
```
Comment (radtriste) [inline Jenkinsfile.deploy]: This one can be removed :)

File: Jenkinsfile.deploy [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            steps {
+                script {
+                    maven.runMavenWithSubmarineSettings('clean deploy', false)
+                }
+            }
+        }
+        stage('Build kogito-runtimes') {
+            steps {
+                checkout([$class: 'GitSCM', branches: [[name: 'master']], browser: [$class: 'GithubWeb', repoUrl: 'git@github.com:kiegroup/kogito-runtimes.git'], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'kogito-runtimes']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'kie-ci-user-key', url: 'git@github.com:kiegroup/kogito-runtimes.git']]])
```
Comment (radtriste) [inline Jenkinsfile.deploy]: No need to checkout kogito-runtimes, it is this repo

**Comment by cristianonicolai** [comment]
@mareknovotny good point. but will address that in another PR.

**Comment by cristianonicolai** [comment]
@mareknovotny @radtriste updated

**Comment by cristianonicolai** [comment]
@mareknovotny I started a thread on Kogito dev list to discuss the versioning. Also, do we need to do something for Jenkins to load this new job?

**Comment by mareknovotny** [comment]
I don't think we need to load it, I saw already it is on there
great so I will read and respond if needed, thanks!

**Pull request merged**